File: src/main/java/com/nerdearla/workshop/validator/BuyerValidator.java
package com.nerdearla.workshop.validator;

import com.nerdearla.workshop.model.Buyer;
import com.nerdearla.workshop.model.error.DisableBuyerError;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

@Component
public class BuyerValidator {

    private static final Logger LOGGER = LoggerFactory.getLogger(BuyerValidator.class);

    public void validate(Buyer buyer) {
        validateIsEnabled(buyer);
    }

    private void validateIsEnabled(Buyer buyer) {
        if (!buyer.getEnabled()) {
            LOGGER.error("buyer {} is disabled", buyer.getId());
            throw new DisableBuyerError();
        }
        LOGGER.info("buyer {} is enabled", buyer.getId());
    }

}

File: src/main/java/com/nerdearla/workshop/validator/BuyerValidator.kt
package com.nerdearla.workshop.validator

import com.nerdearla.workshop.dto.buyer.Buyer
import org.springframework.stereotype.Component
import kotlin.RuntimeException

@Component
class BuyerValidator {

    fun validate(buyer: Buyer, identification: String, gender: String) {
        when {
            !buyer.enabled -> throwBuyerNotEnabled()
            (buyer.identification != identification || buyer.gender != gender) -> throwInvalidBuyer()
        }
    }

    private fun throwBuyerNotEnabled() {
        throw RuntimeException("Buyer is not enabled").also {
            // TODO: log error
        }
    }

    private fun throwInvalidBuyer() {
        throw RuntimeException("Invalid buyer").also {
            // TODO: log error
        }
    }
}